chr/make_chr.py

```diff
@@ -1,100 +1,77 @@
 """Convert an image to NES CHR tile data.
 
-The input image must be 128 pixels wide and use exactly 4 colors when converted
-to grayscale.  Darker colors in the source image will be given lower palette
-indexes in the tile data. (TODO: make this customizable)
+For the most intuitive, consistent results, the input file should be 128x128,
+and each 8x8 tile should contain no more than four colors, with either 
+transparency (counted as a color if present) or black being interpreted as 
+transparent areas of the tile.
+
+Complete 8x8 tiles will be copied starting from the top left of the image in 
+row-major order. Partial tiles at the right or bottom will be ignored. Each 
+tile will be converted to 4 colors (or 3 if the file supports transparency)
+and sorted into ascending-value order. This will cause the darkest color in 
+each tile to be treated as transparent. If the tile contained more colors, 
+the exact collapse into 4 or 3 colors is determined by PIL.
+
+If the file contains fewer than 256 complete 8x8 tiles, the output file will 
+be padded with zeroes to make it 4k long; if it contains more, the first 256 
+will be used and a warning will be issued.
```

Comment:
I've added the infrastructure for padding, cutoff, and scan order, but the default values produce no padding and no limit. Command-line options can be added as needed.

I have added a warning if the file contains more than 4K, but it will still convert the entire file.